### DISCUSSION TITLE
Issues with install

### DISCUSSION THREAD

**Discussion by Captain-Bacon** [soten355/MetalDiffusion#1] [General]
I'm getting the following on install - perhaps the most recent update had a bug?  I followed the instructions to the letter, and it all worked...right up until this point!  I'm not a c0der, so apologies if it's a 'me' thing...

Loading modules...

...system modules loaded...
Traceback (most recent call last):
  File "/Users/[user]/stable-diffusion-webui/stable-diffusion-tensorflow-IntelMetal/dream.py", line 19, in <module>
    from stable_diffusion_tf.stable_diffusion import StableDiffusion
  File "/Users/[user]/stable-diffusion-webui/stable-diffusion-tensorflow-IntelMetal/stable_diffusion_tf/stable_diffusion.py", line 27, in <module>
    import tensorflow as tf
ModuleNotFoundError: No module named 'tensorflow'

**Comment by djr255** [comment]
might be a Tensorflow support issue. I got this error during install:

```
ERROR: Could not find a version that satisfies the requirement tensorflow-macos (from versions: none)
ERROR: No matching distribution found for tensorflow-macos
```

**Reply by Oddj0b** [reply]
~I fixed it by following this guide: https://developer.apple.com/metal/tensorflow-plugin/~

Ignore that above, I'm not able to start Stable diffusion.

**Comment by lifesblackblood** [comment]
I was able to install ([this version](https://github.com/soten355/stable-diffusion-tensorflow-IntelMetal/blob/master/README.md)) on Intel Mac OS 12.6.2, but having trouble launching it. I'm too new to this to know how to troubleshoot what's going wrong. 

> python dream.py
> Loading modules...
> 
> ...system modules loaded...
> Metal device set to: AMD Radeon Pro 555X
> 
> systemMemory: 16.00 GB
> maxCacheSize: 1.00 GB
> 
> ...Stable Diffusion module loaded...
> ...TensorFlow module loaded...
> ...image modules loaded...
> ...WebUI module loaded...
> ...all modules loaded!
>  
> Creating global variables... 
> Loading user preferences...
> 'Settings'
> File does not exist!
> No user preferences found.
> Creating new preferences file.
> Type of variable: 
> <class 'configparser.SectionProxy'>
> Loading prompts...
> ...loaded!
> ...global variables created!
>  
> Creating classes... 
> ...done!
>  
> Searching for diffusion models... 
> 
> Searching for: .ckpt
> ...finished!
> 
> Searching for: .pth
> ...finished!
> 
> Searching for: Keras .h5
> ...finished!
>  
> Searching for VAE models... 
> 
> Searching for: .ckpt
> ...finished!
>  
> Searching for text embeddings... 
> 
> Searching for: .pt
> ...finished!
> 
> Searching for: .bin
> ...finished!
>   
> Starting program: 
> Traceback (most recent call last):
>   File "/Users/apo11/github/stable-diffusion-tensorflow-IntelMetal/dream.py", line 938, in <module>
>     defaultModelValue = modelsWeights[0]
> IndexError: list index out of range
> 

Any suggestions? Thanks in advance

**Reply by digital-pers0n** [reply]
You need to download a model and copy it into the /models/ folder

**Comment by djr255** [comment]
Try prepending this to step 4.

`SYSTEM_VERSION_COMPAT=0 `

**Comment by sevichenko** [comment]
Sorry for my newbie questions, but I already install it but I'm getting this error:
FileNotFoundError: [Errno 2] Unable to open file (unable to open file: name = 'models/Stable-diffusion/text_encoder.h5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)

I searched over the models folder and in the readme file there is these lines:
Within that folder, the program is looking for these four ".h5" files:

1) decoder.h5
2) diffusion_model.h5
3) encoder.h5
4) text_encoder.h5

But in the folder there is no files, I'm looking over google to see if i find something and I can download and put the files on the folder, but I'm not able to find nothing.

Sorry for my poor English, I'm from Spain.

**Comment by StudioNopqrst** [comment]
Followed upon the installation steps, but returned error:
ModuleNotFoundError: No module named 'tensorflow'

Mac M3 Max 96GB RAM Sonoma 14.1

Can anyone advice pls?
